probe-rs/src/probe/stlink/mod.rs

```diff
@@ -1444,6 +1446,10 @@ impl DapAccess for StlinkArmDebug {
 
         Ok(())
     }
+
+    fn probe(&self) -> &dyn DapProbe {
+        unimplemented!()
```

Comment:
We should not panic in such accessible methods. It would be better to rename this function to `try_dap_probe` or similar which can return an error.

Reply:
I really wanted this to return a `&dyn DebugProbe`, but that requires trait upcasting which was just stabilized last week.

Sure, let me rename that.

Reply:
Ah, I see. I fear that for ST-Links which still have a huge userbase, we just panic :/

Reply:
Well, in twelve weeks we can replace it with `try_debug_probe(&self) -> Option<&dyn DebugProbe>;` and it'll suddenly start working for ST-Links! Where it would never have previously worked, since the MCX code assumed a CMSIS-DAP device.